### PR TITLE
Opaque Pointers: Update RayTracingBuiltIn

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**
Create ptr for opaque pointers instead of
typed pointer.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7684bd688d4e51428bdb12651046ad6f5e13835b
<details><summary>CTS tests (Failed: 1/207247)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38904/70464 (55.2%)</li>
<li><details><summary>Failed: 1/70464 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.ray_tracing_pipeline.large_shader_set.gpu.4096
Stack trace: Flake
```
</p></details></li>

<li>Not Supported: 31559/70464 (44.8%)</li><li>Warnings: 0/70464 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70472 (50.7%)</li>
<li>Failed: 0/70472 (0.0%)</li>
<li>Not Supported: 34722/70472 (49.3%)</li><li>Warnings: 0/70472 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66311 (50.5%)</li>
<li>Failed: 0/66311 (0.0%)</li>
<li>Not Supported: 32835/66311 (49.5%)</li><li>Warnings: 0/66311 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/lower/llpcSpirvLowerRayTracingBuiltIn.cpp [hunk 1]
```diff
@@ -165,7 +165,13 @@ bool SpirvLowerRayTracingBuiltIn::runImpl(Module &module) {
...
         if (isa<PointerType>(instTy) && instTy->getPointerAddressSpace() == SPIRAS_Input) {
           assert(isa<GetElementPtrInst>(inst) || isa<BitCastInst>(inst));
-          Type *newInstTy = PointerType::get(instTy->getContainedType(0), SPIRAS_Private);
+          // TODO: Remove this when LLPC will switch fully to opaque pointers.
+          // After transition 'else' will be removed.
+          Type *newInstTy = nullptr;
+          if (instTy->isOpaquePointerTy())
+            newInstTy = PointerType::get(m_builder->getContext(), SPIRAS_Private);
+          else
+            newInstTy = PointerType::get(instTy->getContainedType(0), SPIRAS_Private);
```
Comment (Flakebi) [inline llpc/lower/llpcSpirvLowerRayTracingBuiltIn.cpp:174]: I think this is equivalent to `PointerType::getWithSamePointeeType(instTy, SPIRAS_Private)`.

Comment (mariusz-sikora-at-amd) [reply llpc/lower/llpcSpirvLowerRayTracingBuiltIn.cpp:174]: Right, thanks !

**Comment by amdvlk-admin** [comment]
### Test summary for commit 23da512923e123c3538b68946010cf5efdca63cb
<details><summary>CTS tests (Failed: 0/207247)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38905/70464 (55.2%)</li>
<li>Failed: 0/70464 (0.0%)</li>
<li>Not Supported: 31559/70464 (44.8%)</li><li>Warnings: 0/70464 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70472 (50.7%)</li>
<li>Failed: 0/70472 (0.0%)</li>
<li>Not Supported: 34722/70472 (49.3%)</li><li>Warnings: 0/70472 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33476/66311 (50.5%)</li>
<li>Failed: 0/66311 (0.0%)</li>
<li>Not Supported: 32835/66311 (49.5%)</li><li>Warnings: 0/66311 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**